docs/serverdev/index.rst

```diff
@@ -0,0 +1,130 @@
+
+Setting up Your Development Environment
+---------------------------------------
+
+Macintosh OSX
+'''''''''''''
+
+Third Party Services
+~~~~~~~~~~~~~~~~~~~~
+To run the basic development server, you need to install both Redis and Postgresql.
+Below are basic commands to install them on OSX. By now, you hopefully have Homebrew installed.
+If you don't run::
+
+    /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+
+.. note:: If you would like to have the following services run when you start your machine,
+  follow the instructions that Homeberw provides at the end of running ``install``.
+
+Redis
+*****
+Redis is a popular in-memory datastore that Sentry uses for queuing and caching mechanisms.
+Run the following to install, configure, and execute Redis as a daemonized server::
+
+    brew install redis
+    redis-server --save --daemonize yes
+
+Postgresql
+**********
+Postgresql is the primary database that Sentry uses for all persistent storage.
+Run the following to install, configure, and execute Postgresql as a daemonized server::
+
+    brew install postgresql
+    pg_ctl -D /usr/local/var/postgres start
+
+Third Party Libraries
+~~~~~~~~~~~~~~~~~~~~~
+Sentry depends on a combination of both Python and JavaScript packages that need to be installed
+in order for the server to function properly. Below are basic commands to install the languages
+and their libraries on OSX.
+
+Python
+******
+While OSX ships with an acceptable version of Python, installing the latest binaries from HomeBrew
+is recommended. In addition, it is highly recommended to separate your libraries per-environment
+with the use of virtual environments. One of the easiest ways to curate your virtual environments is
+with `virtualenvwrapper <https://virtualenvwrapper.readthedocs.org/>`_. Run the following to install
+and configure brewed Python and virtualenvwrapper::
+
+    brew install python
+    pip install virtualenvwrapper
+    touch ~/.bashrc >> source /usr/local/bin/virtualenvwrapper.sh
+    source ~/.bashrc
+    mkvirtualenv sentry
+
+JavaScript
+**********
+JavaScript is a tad easier because it separates environments already. To install node.js and
+the Node Package Manager(npm) run::
+
+    brew install nodejs
+
+Installing Libraries
+********************
+Once all of your dependencies are installed, head over to the root of ``getsentry/sentry``.
+Run the following to install both the Python and JavaScript libraries that Sentry depends on
+and some extra pieces that hold the development environment together::
+
+    make develop
+
+Running the Development Server
+------------------------------
+Before you are able to run the development server, you first must create a proper database
+for it to use. Running the following will create the proper database and fill it with example
+data::
+
+    createdb -E utf-8 sentry
+    sentry init
+    sentry upgrade
+    ./bin/load-mocks
```

Comment:
It might be worth explaining the mocks separately (and noting that they're optional.)